### PR TITLE
Activity refactors

### PR DESCRIPTION
**Pull request by andrewckinstler**
#### What's this PR do?
I found that a lot of the methods in this class were accomplishing, basically, the same task, so I made a couple of functions that were less redundant. 
@jpweist @brittanystoroz @khalidwilliams 
#### Where should the reviewer start?
This is all within the Activity class and the major revisions lie around lines 10, 33, and 50.
How should this be manually tested?
I still need to rework the tests for the class, but all of the charts are showing the appropriate data. 
#### What are the relevant tickets?
This will affect tickets 1, 2, and 3, but I don't think I'm quite ready to close them just yet.

### CODE REVIEW

File: src/Activity-Repository.js [hunk 1]
```diff
@@ -7,20 +7,24 @@ class Activity {
     return this.activityData.filter((activityObj) => activityObj.userID === userId);
   }
 
+  findUserCurDate(date) {
+    return this.activityData.filter((activityObj) => activityObj.date === date);
+  }
+
   returnAvgStairsClimbedAllUsersByDate(date)  {
-    let allUsersStairs = this.activityData.filter((activityObj) => activityObj.date === date);
```
Comment (jpweist) [inline src/Activity-Repository.js]: that is so much cleaner

File: test/activity-repository-test.js [hunk 2]
```diff
@@ -11,9 +11,9 @@ describe('Activity', () => {
   beforeEach(() => {
     activity = new Activity(activityTestData);
     user1 = new User(userTestData[0]);
-    user2 = new User(userTestData[1]);
```
Comment (jpweist) [inline test/activity-repository-test.js]: did we need to change some tests?

File: src/Activity-Repository.js [hunk 3]
```diff
@@ -11,24 +11,12 @@ class Activity {
...
-    let allUsersStairs = this.findUserCurDate(date);
-    let allUsersStairsTotal = allUsersStairs.reduce((activityObjA, activityObjB) => activityObjA + activityObjB.flightsOfStairs, 0);
-    return parseInt(allUsersStairsTotal / allUsersStairs.length);
+  returnUserAvgsByDate(date, key)  {
+    let allUsersActivity = this.findUserCurDate(date);
+    let allUsersTotal = allUsersActivity.reduce((activityObjA, activityObjB) => activityObjA + activityObjB[key], 0);
+    return parseInt(allUsersTotal / allUsersActivity.length);
   }
 
-  returnAvgStepsTakenAllUsersByDate(date) {
```
Comment (jpweist) [inline src/Activity-Repository.js]: wow you were able to cut out alot. On my Friends: Weekly Step Challenge all the user data is the same. can you check that

File: src/scripts.js [hunk 4]
```diff
@@ -61,9 +61,9 @@ $('#user-current-stairs-climbed').text(activity.returnStairsClimbedByDate(user.i
 $('#all-users-average-stairs-climbed').text(activity.returnUserAvgsByDate(currentDate, 'flightsOfStairs'));
 $('#user-current-active-mins').text(activity.returnActiveMinutesByDate(user.id, currentDate));
 $('#all-users-average-active-mins').text(activity.returnUserAvgsByDate(currentDate, 'minutesActive'));
-$('#user-step-count-by-week').text(activity.returnNumberOfStepsByWeek(user.id, currentDate))
-$('#user-stairs-climbed-by-week').text(activity.returnStairsClimbedByWeek(user.id, currentDate))
-$('#user-mins-active-by-week').text(activity.returnActiveMinutesByWeek(user.id, currentDate))
+$('#user-step-count-by-week').text(activity.returnUserActivityByWeek(user.id, currentDate, 'numSteps'))
```
Comment (jpweist) [inline src/scripts.js]: hey not sure if this is were the same data issue is. for Friends: Weekly Step Challenge

File: src/scripts.js [hunk 5]
```diff
@@ -101,7 +101,7 @@ function populateFriends(userFriends) {
     return ({
       id: userFriend.id, 
       name: userFriend.returnUserFirstName(),
-      steps: (activity.returnNumberOfStepsByWeek(userFriend.id, currentDate)).reduce((acc, day) => acc += day)})
+      steps: (activity.returnUserActivityByWeek(user.id, currentDate, 'numSteps')).reduce((acc, day) => acc += day)})
```
Comment (jpweist) [inline src/scripts.js]: should this be userFriend?

File: test/activity-repository-test.js (hunk 2 above)
Comment (andrewckinstler) [reply test/activity-repository-test.js]: Yeah, I still have a little ways to go in the testing realm, but it should be as simple as renaming some methods. I might be able to cut down on some tests as well, or at least use some nested `describe` blocks so things are a bit more readable!

File: src/Activity-Repository.js (hunk 3 above)
Comment (andrewckinstler) [reply src/Activity-Repository.js]: Thanks for the heads up here, I did some copy/pasting and apparently my eyes weren't totally open on that one. Fixed now, thanks for the extra set of eyes!

**Comment by brittanystoroz** [comment]
Yes I would definitely update any tests that are breaking due to these changes before merging this PR. You do not want to merge code whose tests are broken.

**Comment by andrewckinstler** [comment]
Fixed up the tests for this class, this should close #1, close #2, close #3, close #5, close #6.

**Comment by jpweist** [comment]
all test pass and friends data is fixed . looks good

**Pull request merged**